bpfd/src/multiprog/tc.rs

```diff
@@ -110,19 +112,72 @@ impl TcDispatcher {
             loader: Some(loader),
         };
         dispatcher.attach_extensions(&mut extensions).await?;
-        dispatcher.attach(old_dispatcher)?;
+        dispatcher.attach(old_dispatcher).await?;
         dispatcher.save()?;
         Ok(dispatcher)
     }
 
-    fn attach(&mut self, old_dispatcher: Option<Dispatcher>) -> Result<(), BpfdError> {
+    /// has_qdisc returns true if the qdisc_name is found on the if_index.
+    async fn has_qdisc(qdisc_name: String, if_index: i32) -> Result<bool, anyhow::Error> {
+        let (connection, handle, _) = rtnetlink::new_connection().unwrap();
+        tokio::spawn(connection);
+
+        let mut qdiscs = handle.qdisc().get().execute();
+        while let Some(qdisc_message) = qdiscs.try_next().await? {
+            if qdisc_message.header.index == if_index {
+                debug!(
+                    "Qdisc: {:?} - Parent: {:?} - Handle: {:?} - Kind: {:?}",
+                    qdisc_message.header.index,
+                    qdisc_message.header.parent,
+                    qdisc_message.header.handle,
+                    qdisc_message.header.info
+                );
+                for nla in qdisc_message.nlas {
+                    if let Nla::Kind(nla_kind) = nla {
+                        if nla_kind == qdisc_name {
+                            return Ok(true);
+                        }
+                    }
+                }
+            }
```

Comment:
I deleted my earlier suggestion.  I talked it over with Dave.  Either would work, but your way has the advantage that it will exit as soon as we find the qdisc in question, so it could exit earlier.  Im making the following new suggestion that just deletes your debug log (I think the logs in the calling code are sufficient) and replaces the for loop with a .contains().

```suggestion
            if qdisc_message.header.index == if_index
                && qdisc_message.nlas.contains(&Nla::Kind(qdisc_name.clone()))
            {
                return Ok(true);
            }
```